crates/core/src/delta_datafusion/find_files/mod.rs

```diff
@@ -1,29 +1,43 @@
 use std::sync::Arc;
 
-use arrow_array::cast::AsArray;
-use arrow_array::{RecordBatch, StringArray};
+use arrow_array::RecordBatch;
 use arrow_schema::SchemaBuilder;
-use arrow_schema::{DataType, Field, Schema};
+use arrow_schema::{ArrowError, DataType, Field, Schema, SchemaRef};
+use arrow_select::concat::concat_batches;
 use async_trait::async_trait;
+use datafusion::datasource::MemTable;
 use datafusion::execution::context::{QueryPlanner, SessionState};
+use datafusion::execution::TaskContext;
+use datafusion::physical_plan::filter::FilterExec;
+use datafusion::physical_plan::limit::LocalLimitExec;
 use datafusion::physical_plan::ExecutionPlan;
 use datafusion::physical_planner::{DefaultPhysicalPlanner, ExtensionPlanner, PhysicalPlanner};
-use datafusion::prelude::{ParquetReadOptions, SessionContext};
+use datafusion::prelude::SessionContext;
 use datafusion_common::{DFSchemaRef, Result, ToDFSchema};
-use datafusion_expr::{Expr, LogicalPlan, UserDefinedLogicalNode};
+use datafusion_expr::{col, Expr, LogicalPlan, UserDefinedLogicalNode};
+use datafusion_physical_expr::create_physical_expr;
 use lazy_static::lazy_static;
 
 use crate::delta_datafusion::find_files::logical::FindFilesNode;
 use crate::delta_datafusion::find_files::physical::FindFilesExec;
-use crate::delta_datafusion::PATH_COLUMN;
+use crate::delta_datafusion::{
+    df_logical_schema, register_store, DeltaScanBuilder, DeltaScanConfigBuilder, PATH_COLUMN,
+};
+use crate::logstore::LogStoreRef;
+use crate::table::state::DeltaTableState;
+use crate::DeltaTableError;
 
 pub mod logical;
 pub mod physical;
 
 lazy_static! {
     static ref ONLY_FILES_SCHEMA: Arc<Schema> = {
         let mut builder = SchemaBuilder::new();
-        builder.push(Field::new(PATH_COLUMN, DataType::Utf8, false));
+        builder.push(Field::new(
+            PATH_COLUMN,
+            DataType::Dictionary(Box::new(DataType::UInt16), Box::new(DataType::Utf8)),
+            false,
+        ));
```

Comment:
Yeah there's definitely a bit of a weird interface mismatch. This operation must only output a file path at most once so there is minimal value in having a dictionary being returned to end user however the dictionary encoding should be used when we perform a scan.
Is it easy to simply have string be returned?

Reply:
I added to delta scan config the ability to turn off the dictionary encoding, it's on by default everywhere else, but here we use just a simple string as you mentioned